api/webhook/validate_auth.go

```diff
@@ -93,63 +88,48 @@ func (v *AuthValidator) Handle(ctx context.Context, req admission.Request) admis
 	}
 	specs := chaos.GetSelectSpec()
 
-	if req.Operation == admissionv1beta1.Update {
-		// when selector is not changed, don't need to validate auth again
-
-		err = v.decoder.DecodeRaw(req.OldObject, oldChaos)
-		if err != nil {
-			return admission.Errored(http.StatusBadRequest, err)
-		}
-
-		oldSpecs := oldChaos.GetSelectSpec()
-		selectSpecChanged := false
-		if len(specs) == len(oldSpecs) {
-			for i, spec := range specs {
-				if !reflect.DeepEqual(oldSpecs[i].GetSelector(), spec.GetSelector()) {
-					selectSpecChanged = true
-					break
-				}
-			}
-		} else {
-			selectSpecChanged = true
-		}
-
-		if !selectSpecChanged {
-			authLog.Info("chaos updated but select spec not changed, auth validate passed")
-			return admission.Allowed("")
-		}
-	}
-
+	requireClusterPrivileges := false
 	affectedNamespaces := make(map[string]struct{})
 
 	for _, spec := range specs {
-		pods, err := selector.SelectPods(context.Background(), v.client, v.reader, spec.GetSelector(), v.clusterScoped, v.targetNamespace, v.allowedNamespaces, v.ignoredNamespaces)
-		if err != nil {
-			return admission.Errored(http.StatusBadRequest, err)
-		}
-
-		for _, pod := range pods {
-			affectedNamespaces[pod.Namespace] = struct{}{}
+		if len(spec.GetSelector().Namespaces) == 0 {
+			requireClusterPrivileges = true
 		}
 
-		// may not exist pod under selector namespace, but still need to validate the privileges
 		for _, namespace := range spec.GetSelector().Namespaces {
 			affectedNamespaces[namespace] = struct{}{}
 		}
 	}
 
-	for namespace := range affectedNamespaces {
-		allow, err := v.auth(username, groups, namespace, chaosKind)
+	if len(affectedNamespaces) > 1 {
+		requireClusterPrivileges = true
+	}
```

Comment:
I think these codes might not be expected. As we already know that the specified affected namespace in `affectedNamespaces`, the scope has been restricted. 

And I think `requireClusterPrivileges = true` means users do not specify about the affected namespace of that chaos experiment, and it seems only happens with `len(SelectorSpec.Namespaces) ==0`.(If there are other situations, it's my carelessness, please let me know. 🙈)

I think it does not need cluster-level permission anymore.

Reply:
update in 7202ec8